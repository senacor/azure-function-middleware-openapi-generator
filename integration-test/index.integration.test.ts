import fs from "node:fs";
import {execSync} from "node:child_process";

describe('azure-function-middleware-openapi-generator', () => {

    beforeAll(() => {
        execSync('cd example && npm run build && npm run generate');
    })

    afterAll(() => {
        execSync('rm example/external.yaml example/internal.json');
    })

    test('generate open api definition for example project internal api in json format', () => {
        const file = fs.readFileSync('example/internal.json').toString();

        expect(file).toMatchSnapshot();
    });

    test('generate open api definition for example project external api in yaml format', () => {
        const file = fs.readFileSync('example/external.yaml').toString();

        expect(file).toMatchSnapshot();
    })
})