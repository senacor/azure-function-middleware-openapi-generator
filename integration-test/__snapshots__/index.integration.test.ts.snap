// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`azure-function-middleware-openapi-generator generate open api definition for example project external api in yaml format 1`] = `
"openapi: 3.0.2
info:
  version: 1.0.0
  title: External API
  description: External API for the shop
servers:
  - url: https://localhost:8080
    description: local server
paths:
  /customer/{customerId}/order:
    post:
      operationId: http-post-order-for-customer
      parameters:
        - name: customerId
          in: path
          required: true
          schema:
            type: string
      requestBody:
        type: object
        properties:
          status:
            type: string
            enum:
              - active
              - finished
          items:
            type: array
            minItems: 1
            items:
              type: object
              properties:
                id:
                  type: string
                  format: uuid
                name:
                  type: string
                price:
                  type: number
              required:
                - id
                - name
                - price
              additionalProperties: false
        required:
          - status
          - items
        additionalProperties: false
      responses:
        "201":
          description: 201 response
          content:
            application/json:
              schema:
                type: object
                properties: {}
                additionalProperties: false
      security:
        - BearerAuth: []
    get:
      operationId: http-get-orders-for-customer
      parameters:
        - name: customerId
          in: path
          required: true
          schema:
            type: string
        - name: status
          in: query
          required: false
          schema:
            type: string
            enum:
              - active
              - finished
      responses:
        "200":
          description: 200 response
          content:
            application/json:
              schema:
                type: array
                items:
                  type: object
                  properties:
                    id:
                      type: string
                      format: uuid
                    status:
                      type: string
                      enum:
                        - active
                        - finished
                    items:
                      type: array
                      minItems: 1
                      items:
                        type: object
                        properties:
                          id:
                            type: string
                            format: uuid
                          name:
                            type: string
                          price:
                            type: number
                        required:
                          - id
                          - name
                          - price
                        additionalProperties: false
                  required:
                    - id
                    - status
                    - items
                  additionalProperties: false
        "404":
          description: 404 response
          content:
            application/json:
              schema:
                type: object
                properties:
                  details:
                    type: string
                required:
                  - details
                additionalProperties: false
      security:
        - BearerAuth: []
components:
  securitySchemes:
    FunctionKeyAuth:
      type: apiKey
      in: header
      name: x-functions-key
    BearerAuth:
      type: http
      scheme: bearer
      bearerFormat: JWT
"
`;

exports[`azure-function-middleware-openapi-generator generate open api definition for example project internal api in json format 1`] = `
"{
  "openapi": "3.0.2",
  "info": {
    "version": "1.0.0",
    "title": "Internal API",
    "description": "Internal API for the shop"
  },
  "servers": [
    {
      "url": "https://localhost:8080",
      "description": "local server"
    }
  ],
  "paths": {
    "/order/{orderId}": {
      "get": {
        "operationId": "internal-http-get-order",
        "parameters": [
          {
            "name": "orderId",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "200 response",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "id": {
                      "type": "string",
                      "format": "uuid"
                    },
                    "status": {
                      "type": "string",
                      "enum": [
                        "active",
                        "finished"
                      ]
                    },
                    "items": {
                      "type": "array",
                      "items": {
                        "type": "object",
                        "properties": {
                          "id": {
                            "type": "string",
                            "format": "uuid"
                          },
                          "name": {
                            "type": "string"
                          },
                          "price": {
                            "type": "number"
                          }
                        },
                        "required": [
                          "id",
                          "name",
                          "price"
                        ],
                        "additionalProperties": false
                      }
                    }
                  },
                  "required": [
                    "id",
                    "status"
                  ],
                  "additionalProperties": false
                }
              }
            }
          },
          "404": {
            "description": "404 response",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "details": {
                      "type": "string"
                    }
                  },
                  "required": [
                    "details"
                  ],
                  "additionalProperties": false
                }
              }
            }
          }
        },
        "security": [
          {
            "FunctionKeyAuth": []
          }
        ]
      }
    }
  },
  "components": {
    "securitySchemes": {
      "FunctionKeyAuth": {
        "type": "apiKey",
        "in": "header",
        "name": "x-functions-key"
      },
      "BearerAuth": {
        "type": "http",
        "scheme": "bearer",
        "bearerFormat": "JWT"
      }
    }
  }
}"
`;
